modules/imgproc/src/smooth.cpp

```diff
@@ -1295,6 +1295,60 @@ struct ColumnSum<int, float> :
 
 #ifdef HAVE_OPENCL
 
+static bool ocl_boxFilter3x3_8UC1( InputArray _src, OutputArray _dst, int ddepth,
+                                   Size ksize, Point anchor, int borderType, bool normalize )
+{
+    const ocl::Device & dev = ocl::Device::getDefault();
+    int type = _src.type(), sdepth = CV_MAT_DEPTH(type), cn = CV_MAT_CN(type);
+
+    if (ddepth < 0)
+        ddepth = sdepth;
+
+    if (anchor.x < 0)
+        anchor.x = ksize.width / 2;
+    if (anchor.y < 0)
+        anchor.y = ksize.height / 2;
+
+    if ( !(dev.isIntel() && (type == CV_8UC1) &&
+         (_src.offset() == 0 && _dst.offset() == 0) &&
+         (_src.step() == _dst.step()) &&
```

Comment:
> _src.step() == _dst.step()

Why do you need this condition?

BTW, I mean this check in the one of my notes:
```
((src.step() % 4) == 0 && (dst.step() % 4) == 0)
```
But second half of this condition must be done **after** this line (and `_dst.offset()` too):
```
_dst.create(size, CV_MAKETYPE(ddepth, cn));
```